xray/core/indexing.py

```diff
@@ -368,7 +368,12 @@ class PandasIndexAdapter(utils.NDArrayMixin):
     def __init__(self, array, dtype=None):
         self.array = utils.safe_cast_to_index(array)
         if dtype is None:
-            dtype = array.dtype
+            # if the index -'hides' its dtype such as
+            # PeriodIndex use an object dtype
+            if self.array._box_scalars:
```

Comment:
A bigger concern is that xray will start crashing for every index type is the `_box_scalers` attribute goes away. At the very least, you need to use `getattr`.